File: ui-testing/selenium-java/src/test/java/pageobjects/SauceDemoLoginPage.java
package pageobjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class SauceDemoLoginPage extends AbstractPage {

    public final static By usernameInput = By.id("user-name");
    public final static By passwordInput = By.id("password");
    public final static By loginButton = By.id("login-button");

    public SauceDemoLoginPage(WebDriver driver) {
        super(driver);
    }

    public void loadPage() {
        getDriver().get("https://www.saucedemo.com");
    }
}

File: ui-testing/cypress-node/cypress/page-objects/sauce-demo-login-page.js
/// <reference types="cypress" />

export class SauceDemoLoginPage {
    loadPage() {
        cy.visit('https://www.saucedemo.com')
    }

    enterCredentials(username, password) {
        cy.get('#user-name').should('be.visible')
            .type(username)
        cy.get('#password')
            .type(password, { log:false })
    }

    clickLoginButton() {
        cy.get('#login-button').should('be.visible')
            .click()
    }
}